sdk/core/Azure.Core/src/Pipeline/RetryPolicy.cs

```diff
@@ -14,30 +14,25 @@ namespace Azure.Core.Pipeline
     /// <summary>
     /// Represents a policy that can be overriden to customize whether or not a request will be retried and how long to wait before retrying.
     /// </summary>
-    internal abstract class RetryPolicy : HttpPipelinePolicy
+    public class RetryPolicy : HttpPipelinePolicy
     {
-        private readonly RetryMode _mode;
-        private readonly TimeSpan _delay;
-        private readonly TimeSpan _maxDelay;
         private readonly int _maxRetries;
+        private readonly DelayStrategy _delayStrategy;
 
-        private readonly Random _random = new ThreadSafeRandom();
-
-        private const string RetryAfterHeaderName = "Retry-After";
-        private const string RetryAfterMsHeaderName = "retry-after-ms";
-        private const string XRetryAfterMsHeaderName = "x-ms-retry-after-ms";
+        /// <summary>
+        ///
+        /// </summary>
+        protected DelayStrategy DelayStrategy => _delayStrategy;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="RetryPolicy"/> class.
         /// </summary>
-        /// <param name="options">The set of options to use for configuring the policy.</param>
-        protected RetryPolicy(RetryOptions? options = default)
+        /// <param name="maxRetries"></param>
+        /// <param name="delayStrategy">The delay strategy to use</param>
+        public RetryPolicy(int? maxRetries = default, DelayStrategy? delayStrategy = default)
```

Comment:
Should we just be explicit about the maxRetries default value?
```suggestion
        public RetryPolicy(int maxRetries = 3, DelayStrategy? delayStrategy = default)
```